core/src/main/scala/org/apache/spark/sql/TiSparkFunctionRegistry.scala

```diff
@@ -0,0 +1,114 @@
+package org.apache.spark.sql
+
+import org.apache.spark.sql.catalyst.analysis.FunctionRegistry.FunctionBuilder
+import org.apache.spark.sql.catalyst.expressions.{Expression, ExpressionDescription, ExpressionInfo, Microsecond, RuntimeReplaceable, TimeToSec, ToDays, ToSeconds, WeekDay, YearWeek}
+
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success, Try}
+
+object TiSparkFunctionRegistry {
+
+// Note: Whenever we add a new entry here, make sure we also update ExpressionToSQLSuite
+  val expressions: Map[String, (ExpressionInfo, FunctionBuilder)] = Map(
+    // misc non-aggregate functions
+    expression[Microsecond]("microsecond"),
+    expression[WeekDay]("weekday"),
+    expression[ToSeconds]("to_seconds"),
+    expression[ToDays]("to_days"),
+    expression[TimeToSec]("time_to_sec"),
+    expression[YearWeek]("yearweek")
+    // TODO: this need import antler 4 into our code base.
+    // the behavior is like the following:
+    // select extract year from '2019-01-10' which is just
```

Comment:
Is this syntax supported in create table? If so, you do need our own antlr parser. If it only appears in select, why do we need this?